In [2]:
from langchain_groq import ChatGroq

llm= ChatGroq(
    model="llama3-70b-8192",
    groq_api_key="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",
    temperature=0,

)

response = llm.invoke("What is the capital of France?")
print(response.content)

The capital of France is Paris.


In [4]:
response = llm.invoke("how to be a data scientist")
print(response.content)

The coveted title of Data Scientist! Here's a comprehensive guide to help you become a data scientist:

**Step 1: Build a strong foundation in math and statistics**

* Linear Algebra: Understand concepts like vector spaces, eigendecomposition, and singular value decomposition.
* Calculus: Familiarize yourself with derivatives, integrals, and optimization techniques.
* Probability and Statistics: Study probability distributions, Bayes' theorem, hypothesis testing, and confidence intervals.
* Mathematical Optimization: Learn about linear and nonlinear programming, dynamic programming, and stochastic optimization.

**Step 2: Learn programming skills**

* Python: Focus on Python 3.x, as it's the most popular language used in data science. Learn libraries like NumPy, Pandas, and scikit-learn.
* R: Familiarize yourself with R, especially if you're interested in statistical modeling and data visualization.
* SQL: Understand how to work with databases, perform queries, and extract data.

**Ste

In [5]:
from langchain_community.document_loaders import WebBaseLoader

loader= WebBaseLoader("https://technopark.org/job-details/18627?job=DATA%20SCIENCE%20&%20AI/ML")
page_data= loader.load().pop().page_content
print(page_data)

USER_AGENT environment variable not set, consider setting it to identify your requests.



















 
















Technopark Job posting for DATA SCIENCE & AI/ML

MenuThis Is TechnoparkCompanies & JobsInfrastructure & Services< All Jobs REIZEND PRIVATE LIMITEDSBC Module 15, 
-2 Floor, Thejaswini Building, 
Technopark phase 1, Trivandrum , 695581http://www.reizendretail.inDATA SCIENCE & AI/MLClosing Date:26,Apr 2025Job Published: 12,Mar 2025Contact Email:  careers@reizendretail.in Brief Description
Location: Trivandrum

Qualifications

Bachelor's or Master's degree in Data Science, Statistics, Mathematics, Computer Science, or a related field.
Experience: 0 to 1 yrs (Freshers can apply)
Package- 4-6 LPA
Preferred Skills
Proficiency in Python programming with a strong grasp of core concepts.
Basic Data Structures and Algorithms: Familiarity with common data structures (lists, dictionaries, tuples) and basic algorithms (sorting, searching) is advantageous.
Problem-Solving Skills: Ability to analyse and solve programming problems using Python.
Understanding of Object

Technopark Job posting for DATA SCIENCE & AI/ML

MenuThis Is TechnoparkCompanies & JobsInfrastructure & Services< All Jobs REIZEND PRIVATE LIMITEDSBC Module 15, 
-2 Floor, Thejaswini Building, 
Technopark phase 1, Trivandrum , 695581http://www.reizendretail.inDATA SCIENCE & AI/MLClosing Date:26,Apr 2025Job Published: 12,Mar 2025Contact Email:  careers@reizendretail.in Brief Description
Location: Trivandrum

Qualifications

Bachelor's or Master's degree in Data Science, Statistics, Mathematics, Computer Science, or a related field.
Experience: 0 to 1 yrs (Freshers can apply)
Package- 4-6 LPA
Preferred Skills
Proficiency in Python programming with a strong grasp of core concepts.
Basic Data Structures and Algorithms: Familiarity with common data structures (lists, dictionaries, tuples) and basic algorithms (sorting, searching) is advantageous.
Problem-Solving Skills: Ability to analyse and solve programming problems using Python.
Understanding of Object-Oriented Programming (OOP): Basic understanding of OOP principles like classes, objects, inheritance, and encapsulation
Basic knowledge in Data Science including but not limited to ETL, EDA, Data Processing, Visualization etc.
Familiarity with machine learning and deep learning models (e.g., TensorFlow, Keras, PyTorch).
Experience working with SQL DMBS like PostgreSQL is a plus.
Basic understanding of Natural Language Processing (NLP) concepts
Knowledge of Python frameworks like Django
Familiarity with Jira and Agile methodology will be an added advantage.
Certification or experience in data analysis, data mining, and statistical modelling is an added advantage
Strong knowledge of statistical concepts and techniques.
Knowledge in data visualization tools (Tableau, Power BI, etc.) and SQL.
Excellent problem-solving and analytical skills.
Strong communication and presentation skills.
Certification in Data Science is an added advantage.
Happening HereAll NewsTendersEventsBlogQuick LinksCompany LoginVisitor PassFacility BookingContact UsSpace Request ApplicationCompanies & JobsBrowse CompaniesCareers@Technopark"A surprising innovation hotspot powered by human spirit and fuelled by a nurturing environment."

In [6]:
from langchain_core.prompts import PromptTemplate

prompt_extract= PromptTemplate.from_template(
    """ 
    ###SCRAPED TEXT: 
    {page_data}
    The text is from the career page of a data science company.
    Extract the job postings and return them in JSON format containing the following fields:
    - job title
    - job description
    - skills required
    - location
    - salary
    - decription
    Only return valid JSON
    ### VALID JSON(NO PREAMBLE TEXT):
    """
)
chain_extract= prompt_extract | llm
res= chain_extract.invoke(input={'page_data':page_data})
print(res.content)


Here is the extracted job posting in JSON format:

```
{
  "jobTitle": "DATA SCIENCE & AI/ML",
  "jobDescription": "Brief Description",
  "skillsRequired": [
    "Proficiency in Python programming",
    "Basic Data Structures and Algorithms",
    "Problem-Solving Skills",
    "Understanding of Object-Oriented Programming (OOP)",
    "Basic knowledge in Data Science",
    "Familiarity with machine learning and deep learning models",
    "Experience working with SQL DMBS",
    "Basic understanding of Natural Language Processing (NLP) concepts",
    "Knowledge of Python frameworks like Django",
    "Familiarity with Jira and Agile methodology",
    "Certification or experience in data analysis, data mining, and statistical modelling",
    "Strong knowledge of statistical concepts and techniques",
    "Knowledge in data visualization tools",
    "Excellent problem-solving and analytical skills",
    "Strong communication and presentation skills",
    "Certification in Data Science"
  ],
  

Here is the extracted job posting in JSON format:

```
{
  "jobTitle": "DATA SCIENCE & AI/ML",
  "jobDescription": "Brief Description",
  "skillsRequired": [
    "Proficiency in Python programming",
    "Basic Data Structures and Algorithms",
    "Problem-Solving Skills",
    "Understanding of Object-Oriented Programming (OOP)",
    "Basic knowledge in Data Science",
    "Familiarity with machine learning and deep learning models",
    "Experience working with SQL DMBS",
    "Basic understanding of Natural Language Processing (NLP) concepts",
    "Knowledge of Python frameworks like Django",
    "Familiarity with Jira and Agile methodology",
    "Certification or experience in data analysis, data mining, and statistical modelling",
    "Strong knowledge of statistical concepts and techniques",
    "Knowledge in data visualization tools",
    "Excellent problem-solving and analytical skills",
    "Strong communication and presentation skills",
    "Certification in Data Science"
  ],
  "location": "Trivandrum",
  "salary": "4-6 LPA",
  "description": "Location: Trivandrum. Qualifications: Bachelor's or Master's degree in Data Science, Statistics, Mathematics, Computer Science, or a related field. Experience: 0 to 1 yrs (Freshers can apply)"
}
```


In [5]:
from langchain_core.output_parsers import JsonOutputParser

json_parser =JsonOutputParser()
parsed_json= json_parser.parse(res.content)
parsed_json

{'jobTitle': 'DATA SCIENCE & AI/ML',
 'jobDescription': 'Brief Description',
 'skillsRequired': ['Proficiency in Python programming',
  'Basic Data Structures and Algorithms',
  'Problem-Solving Skills',
  'Understanding of Object-Oriented Programming (OOP)',
  'Basic knowledge in Data Science',
  'Familiarity with machine learning and deep learning models',
  'Experience working with SQL DMBS',
  'Basic understanding of Natural Language Processing (NLP) concepts',
  'Knowledge of Python frameworks like Django',
  'Familiarity with Jira and Agile methodology',
  'Certification or experience in data analysis, data mining, and statistical modelling',
  'Strong knowledge of statistical concepts and techniques',
  'Knowledge in data visualization tools',
  'Excellent problem-solving and analytical skills',
  'Strong communication and presentation skills',
  'Certification in Data Science'],
 'location': 'Trivandrum',
 'salary': '4-6 LPA',
 'description': "Location: Trivandrum. Qualificatio

In [17]:
prompt_email= PromptTemplate.from_template(
    """
    ### JOB DESCRIPTION:
    {job_description}

    ### INSTRUCTIONS:
    Create a medium sized cold email based on my portfolio and resume in github(https://github.com/Abhiram-who)
    attach the resume link and portfolio link in the email.
    I am an aspiring data scientist with a strong background in statistics, machine learning, deep learning and genai.
    make use of my skills and experience from my projects.
    """
)

chain_email= prompt_email | llm
email= chain_email.invoke(input={'job_description':parsed_json['jobDescription']})
print(email.content)

Here's a medium-sized cold email based on your portfolio and resume on GitHub:

**Subject: Data Science Expertise for Your Business Needs**

Dear [Hiring Manager's Name],

I came across your company, [Company Name], and was impressed by the innovative work you're doing in [industry/field]. As an aspiring data scientist with a strong background in statistics, machine learning, deep learning, and generative AI, I believe I can contribute to your team's success.

With a solid foundation in mathematical modeling and a passion for extracting insights from complex data, I've developed a range of skills that can help drive business growth. My projects on GitHub (https://github.com/Abhiram-who) showcase my expertise in:

* **Predictive Modeling**: Developed a credit risk assessment model using logistic regression and decision trees, achieving an accuracy of 92%.
* **Deep Learning**: Built a convolutional neural network for image classification, achieving a test accuracy of 95%.
* **Generative 

Here's a medium-sized cold email based on your portfolio and resume on GitHub:

**Subject: Data Science Expertise for Your Business Needs**

Dear [Hiring Manager's Name],

I came across your company, [Company Name], and was impressed by the innovative work you're doing in [industry/field]. As an aspiring data scientist with a strong background in statistics, machine learning, deep learning, and generative AI, I believe I can contribute to your team's success.

With a solid foundation in mathematical modeling and a passion for extracting insights from complex data, I've developed a range of skills that can help drive business growth. My projects on GitHub (https://github.com/Abhiram-who) showcase my expertise in:

* **Predictive Modeling**: Developed a credit risk assessment model using logistic regression and decision trees, achieving an accuracy of 92%.
* **Deep Learning**: Built a convolutional neural network for image classification, achieving a test accuracy of 95%.
* **Generative AI**: Created a generative adversarial network (GAN) for generating synthetic data, demonstrating potential applications in data augmentation.

My resume (https://github.com/Abhiram-who/blob/main/Resume.pdf) provides more details on my education, skills, and experience. I'm confident that my analytical mindset and technical skills can help your organization make data-driven decisions.

If you're looking for a dedicated and enthusiastic data scientist to join your team, I'd love to discuss how my skills can align with your business needs. Please feel free to reach out to me at [Your Email Address].

Best regards,

Abhiram

**P.S.** I'd appreciate the opportunity to explore potential collaboration opportunities or discuss how my skills can contribute to your organization's success.

This email aims to:

1. Introduce yourself and your background in data science.
2. Highlight your relevant skills and experience through your projects on GitHub.
3. Showcase your achievements and expertise in specific areas (predictive modeling, deep learning, and generative AI).
4. Encourage the hiring manager to review your resume and portfolio.
5. Express your enthusiasm for potential collaboration opportunities.

Remember to customize the email by replacing [Hiring Manager's Name], [Company Name], and [industry/field] with the actual information of the company and hiring manager you're targeting.
